### DISCUSSION TITLE
How to start a new span with given trace Id in a console application

### DISCUSSION THREAD

**Discussion by kushal-ti** [open-telemetry/opentelemetry-java#4579] [Q&A]
I have a web application that responds to a request by starting up a background process. The background process has logs of it's own that are also sent to the same log sink. 
I want to export traces and logs from the background process to my collector, meaning i want the traces to have the same trace id as my web application.
How can i start a span in my background process and then set the trace id to the incoming trace id from the web application. Assume that the trace id and span id will be passed to the background process as strings

**Comment by jkwatson** [comment]
I think in this case, you'll need to mimic what the standard propagators do. That is, create a SpanContext from the traceid/spanid and put it into a Context. Take a look at the propagators from how this works.

**Reply by irajhedayati** [reply]
The propagators work if only both traceid/spanid are provided. How about when you have just the trace ID? I know that Go language they can create spans just using the trace ID but I can't figure out how to do it in Java
